opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -3693,60 +3720,78 @@ export const elPaginate = async (context, user, indexName, options = {}) => {
     query.docvalue_fields = REL_DEFAULT_FETCH;
   }
   logApp.debug('[SEARCH] paginate', { query });
-  return elRawSearch(context, user, types !== null ? types : 'Any', query)
-    .then((data) => {
-      return buildSearchResult(data, first, body.search_after, connectionFormat);
-    })
-    .catch(
-      /* v8 ignore next */ (err) => {
-        const root_cause = err.meta?.body?.error?.caused_by?.type;
-        if (root_cause === TOO_MANY_CLAUSES) throw ComplexSearchError();
-        throw DatabaseError('Fail to execute engine pagination', { cause: err, root_cause, query, queryArguments: options });
-      }
-    );
+  try {
+    const data = await elRawSearch(context, user, types !== null ? types : 'Any', query);
+    const parsedResult = await buildSearchResult(context, user, data, first, body.search_after, options);
+    return withResultMeta ? parsedResult : parsedResult.elements;
+  } catch (err) {
+    const root_cause = err.meta?.body?.error?.caused_by?.type;
+    if (root_cause === TOO_MANY_CLAUSES) throw ComplexSearchError();
+    throw DatabaseError('Fail to execute engine pagination', { cause: err, root_cause, query, queryArguments: options });
+  }
 };
-export const elList = async (context, user, indexName, opts = {}) => {
-  const { maxSize = undefined, logForMigration = false } = opts;
+
+const elListRaw = async (context, user, indexName, opts = {}) => {
```

Comment:
to be renamed to elRepaginate ? (eg it rebuild page with full required size from page which may have been post filtered and smaller than the required size)